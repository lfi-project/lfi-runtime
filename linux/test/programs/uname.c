#define _GNU_SOURCE
#include <stdio.h>
#include <sys/syscall.h>
#include <sys/utsname.h>
#include <unistd.h>

int
main()
{
    struct utsname buf;

    // Use the syscall directly instead of uname() wrapper
    long result = syscall(SYS_uname, &buf);

    if (result == -1) {
        perror("syscall SYS_uname failed");
        return 1;
    }

    printf("System name:    %s\n", buf.sysname);
    printf("Node name:      %s\n", buf.nodename);
    printf("Release:        %s\n", buf.release);
    printf("Version:        %s\n", buf.version);

    return 0;
}
