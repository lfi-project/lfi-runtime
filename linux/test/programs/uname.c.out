System name:    Linux LFI
Node name:      
Release:        4.5.0-lfi
Version:        0.0.0-unknown
